cpu/cortexm_common/Makefile.dep

```diff
@@ -32,3 +32,6 @@ endif
 ifeq (1, $(DEVELHELP))
   FEATURES_OPTIONAL += cortexm_mpu
 endif
+
+# Make calls to malloc and friends thread-safe
+USEMODULE += malloc_thread_safe
```

Comment:
Is this module optional?

Reply:
In a sense that it is not needed by every architecture: Yes.

In a sense that an application developer should be able to opt out of it fro e.g. a Cortex M board: No. I don't think we need an "Do you want random data corruptions and hardfaults" checkbox exposed :-)

People not using `malloc()` and friends will only be affected by the additional time needed to compile this. The functions and the mutex will be garbage collected by the linker, if unused.

Reply:
> In a sense that an application developer should be able to opt out of it fro e.g. a Cortex M board: No. I don't think we need an "Do you want random data corruptions and hardfaults" checkbox exposed :-)

My thoughts exactly. So far I see cortexm and atmega using this module, and that's mandatory. Is it also the case for the remaining architectures that you mention on the PR description? If this actually depends on whether or not the architecture needs the module, then I think it should not be configurable by the user. In that case we should consider removing the prompt from the Kconfig module symbol.

Reply:
> If this actually depends on whether or not the architecture needs the module, then I think it should not be configurable by the user.

Agreed.

> In that case we should consider removing the prompt from the Kconfig module symbol.

Let me check if I can figure out how to do that.

Reply:
Just removing the text after `bool` seems to do the trick. Is that the supposed way to do it?

Is the `depends on TEST_KCONFIG` btw. correct? I just copied it in a cargo cult manner. I guess when this is not chosen by end users anyway, dependencies should not added, right? So I should remove that?

Reply:
> Just removing the text after `bool` seems to do the trick. Is that the supposed way to do it?

Yes, that is OK.

> Is the `depends on TEST_KCONFIG` btw. correct? I just copied it in a cargo cult manner. I guess when this is not chosen by end users anyway, dependencies should not added, right? So I should remove that?

The dependency is correct. It is being used so the `MODULE_` symbols do not appear in the normal configuration while migrating. It's true that this symbol is hidden now and does not really affect the dependency resolution when `TEST_KCONFIG` is not set. 

I see two solutions: 1) Remove the dependency altogether, as you suggest. 2) Conditionally `select` this symbol from the architectures (i.e. `select MODULE_MALLOC_THREAD_SAFE if TEST_KCONFIG`) and also keep the dependency. IMO 2) is a bit more consistent with the current approach where all the `MODULE_` symbols are part of the migration and should depend on `TEST_KCONFIG`.